scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java

```diff
@@ -94,17 +95,24 @@ public static Serializable convertByte2Object(final byte[] value) {
         Future<Map<K, V>> future = executor.submit(callable);
 
         try {
-            if (totalSize < deserializationThreshold) {
-                // If the amount of bytes to deserialize is small, the deserialization should not exceed a timeout,
-                // otherwise it means a deadlock occurs (most likely due to jython types)
-                // timeout is disabled for serialization
-                resultMap = future.get(secondsToWait, TimeUnit.SECONDS);
-            } else {
+            try {
+                if (totalSize < deserializationThreshold) {
+                    // If the amount of bytes to deserialize is small, the deserialization should not exceed a timeout,
+                    // otherwise it means a deadlock occurs (most likely due to jython types)
+                    // timeout is disabled for serialization
+                    resultMap = future.get(secondsToWait, TimeUnit.SECONDS);
+                } else {
+                    resultMap = future.get();
+                }
+            } catch (TimeoutException e) {
+                if (StackTraceUtil.getAllStackTraces().contains("org.python.core")) {
```

Comment:
dangerous to put only "contain org.python.core" maybe put more explicit message to check. Then, I don't see python-related messages in https://trello.com/c/uCAczDYP/5374-tryqa-deserialization-problem

Reply:
I think this is less dangerous than what we have now. we want to stop the scheduler only when the specific python deadlock occur.
But you are right the error message must be changed.